src/Framework/EventGen/GMCJDriver.cxx

```diff
@@ -641,13 +679,23 @@ void GMCJDriver::ComputeProbScales(void)
     fPmax.clear();
   }
 
-  // for maximum interaction probability vs E /for given geometry/ I will
-  // be using 300 bins up to the maximum energy for the input flux
-  // double de   = fEmax/300.;//djk june 5, 2013
-  double de   = fEmax/300.;//djk june 5, 2013
-  double emin = 0.0;
-  double emax = fEmax + de;
-  int n = 1 + (int) ((emax-emin)/de);
+  int n = 0;
+  double * ebins = new double[400];
+  if (fPmaxLogBinning) {
+    n = 300;
+    double emin = 0.1;
+    double de = (TMath::Log10(fEmax) - TMath::Log10(emin)) / n;
+    for(int i=0; i<=n; i++) ebins[i] = TMath::Power(10., TMath::Log10(emin) + i * de);
+  }
+  else {
+    // for maximum interaction probability vs E /for given geometry/ I will
+    // be using 300 bins up to the maximum energy for the input flux
+    double de   = fEmax/300.;//djk june 5, 2013
+    double emin = 0.0;
+    double emax = fEmax + de;
+    n = 1 + (int) ((emax-emin)/de);
+    for(int i=0; i<n; i++) ebins[i] = emin + i * (emax-emin)/n;
+  }
```

Comment:
I know `300` is in the original code, but could this also be another parameter (with the default 300) and this size should be used to allocate the `ebins` array.  But first the filling of `ebins` goes one beyond `300` .e.g.  `(int i=0; i<=n; i++)`  which the saving grace is that `bins` is allocated to be `[400]` (could this be `[n+1]` if `n=300`?).  I'm presuming that `n=`+(int)((emax-emin)/de) is going to give `301`

Also verify that `ebins` is deleted somewhere later (i.e. not a memory leak)

Reply:
fixed